Sources/Purchasing/Purchases/PurchaseParams.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchaseParams.swift
+//
+//  Created by mark on 17/10/24.
+
+import Foundation
+
+#if ENABLE_PURCHASE_PARAMS
+
+/**
+ * ``PurchaseParams`` can be used to add configuration options when making a purchase.
+ * This class follows the builder pattern.
+ *
+ * Example making a purchase using ``PurchaseParams``:
+ *
+ * ```swift
+ * let params = PurchaseParams.Builder()
+ *                            .with(metadata: ["key": "value"])
+ *                            .with(promotionalOffer: promotionalOffer)
+ *                            .build()
+ *  Purchases.shared.purchase(package: package, params: params)
+ * ```
+ */
+@objc(RCPurchaseParams) public final class PurchaseParams: NSObject {
```

Comment:
Opted for using the builder pattern instead of using a Set like in StoreKit's [PurchaseOption](https://developer.apple.com/documentation/storekit/product/purchaseoption) as well as calling it `PurchaseParams` so it more closely resembles our Android APIs. Additionally, we already use the builder pattern in the `Configuration` class so it's consistent.